content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/__methods__/wait_for_completion.rb

```diff
@@ -0,0 +1,64 @@
+#
+# Description: This class checks to see if the external provision has completed
+#
+module ManageIQ
+  module Automate
+    module Service
+      module Generic
+        module StateMachines
+          module GenericLifecycle
+            class WaitForCompletion
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                @handle.log("info", "Starting wait_for_completion")
+
+                task = @handle.root["service_template_provision_task"]
+                service = task.try(:destination)
+
+                unless service
+                  @handle.log(:error, 'Service is nil')
+                  raise 'Service is nil'
+                end
+
+                wait_for_completion(service)
+
+                unless @handle.root['ae_reason'].blank?
+                  task.miq_request.user_message = @handle.root['ae_reason']
+                end
+                  @handle.root['ae_result'] = 'ok'
+                @handle.log("info", "Ending wait_for_execute_external")
+
+              end
+
+              private
+
+              def wait_for_completion(service)
+                status, reason = status, reason = service.wait_for_completion(@handle.root["service_context"])
```

Comment:
@bzwei @mkanoor and I discussed the name this morning and determined that it was okay because we're not doing the internal methods. I'm open to changing it though. Can you think of a better name?